store/tikv/region_cache.go

```diff
@@ -57,69 +57,72 @@ func (c *RegionCache) DropRegion(regionID uint64) {
 	delete(c.regions, regionID)
 }
 
-// NextStore picks next store as new leader, if out of range of stores delete region.
-func (c *RegionCache) NextStore(regionID uint64) {
+// NextPeer picks next peer as new leader, if out of range of peers delete region.
+func (c *RegionCache) NextPeer(regionID uint64) {
 	// A and B get the same region and current leader is 1, they both will pick
-	// store 2 as leader.
+	// peer 2 as leader.
 	c.mu.RLock()
 	region, ok := c.regions[regionID]
 	c.mu.RUnlock()
 	if !ok {
 		return
 	}
-	if leader, err := region.NextStore(); err != nil {
+	if leader, err := region.NextPeer(); err != nil {
 		c.mu.Lock()
 		delete(c.regions, regionID)
 		c.mu.Unlock()
 	} else {
-		c.UpdateLeader(regionID, leader)
+		c.UpdateLeader(regionID, leader.GetId())
 	}
 }
 
-// UpdateLeader update some region cache with newer leader store ID.
-func (c *RegionCache) UpdateLeader(regionID, leaderStoreID uint64) {
+// UpdateLeader update some region cache with newer leader info.
+func (c *RegionCache) UpdateLeader(regionID, leaderID uint64) {
 	c.mu.RLock()
 	old, ok := c.regions[regionID]
 	c.mu.RUnlock()
 	if !ok {
-		log.Debugf("regionCache: cannot find region when updating leader %d,%d", regionID, leaderStoreID)
+		log.Debugf("regionCache: cannot find region when updating leader %d,%d", regionID, leaderID)
 		return
 	}
 	var (
+		peer  *metapb.Peer
 		store *metapb.Store
 		err   error
 	)
 
-	curStoreIdx := -1
-	for idx, storeID := range old.meta.StoreIds {
-		if storeID == leaderStoreID {
+	curPeerIdx := -1
+	for idx, p := range old.meta.Peers {
+		if p.GetId() == leaderID {
+			peer = p
 			// No need update leader.
-			if idx == old.curStoreIdx {
+			if idx == old.curPeerIdx {
 				return
 			}
-			curStoreIdx = idx
+			curPeerIdx = idx
 			break
 		}
 	}
-	if curStoreIdx != -1 {
-		store, err = c.pdClient.GetStore(leaderStoreID)
+	if peer != nil {
+		store, err = c.pdClient.GetStore(peer.GetStoreId())
 	}
 
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	delete(c.regions, regionID)
 
-	if curStoreIdx == -1 || err != nil {
-		// Can't find the store in cache, or error occurs when loading
+	if peer == nil || err != nil {
+		// Can't find the peer in cache, or error occurs when loading
 		// store from PD.
```

Comment:
@shenli It's not a typo here. `err == nil` means L107 failed load Store from PD.